backends/dpdk/dpdkHelpers.cpp

```diff
@@ -786,24 +786,49 @@ bool ConvertStatementToDpdk::preorder(const IR::MethodCallStatement *s) {
     return false;
 }
 
-// This function assumes EliminateSwitch midend pass is applied and only handles
-// action_run in switch expression.
+// If we get a switch statement with constant switch expression, only emit the body of
+// matching case label
+bool ConvertStatementToDpdk::handleConstSwitch(const IR::SwitchStatement *s) {
```

Comment:
I think we already have this optimization in the compiler, don't we?
Do you see this triggered?

Reply:
Yes. SimplifySwitch works on optimizing the constant expression in switch.
However, this case is hitting after Midend does copy propagation. Calling the SimplifySwitch pass again after local copy propagation and constant folding works. Shall I update the PR with this change?

Reply:
Should be shorter too

Reply:
This isn't working as expected for certain cases. I will update the PR after a little more testing.

Reply:
@mbudiu-vmw I updated the code to use the SimplifySwitch pass. 
Updated the SimplifySwitch pass to handle the case where the matching case label does not have associated block statement body and should fallthrough.

```
switch(16) {
    case 16:
    case 32: {   // block statement to be executed for both case 16 and case 32}
}
```

Please take a look.

Reply:
Is this a bug in the original code?